src/Val.h

```diff
@@ -155,7 +155,7 @@ class Val : public Obj
 
 	UNDERLYING_ACCESSOR_DECL(detail::IntValImplementation, zeek_int_t, AsInt)
 	UNDERLYING_ACCESSOR_DECL(BoolVal, bool, AsBool)
-	UNDERLYING_ACCESSOR_DECL(EnumVal, int, AsEnum)
+	UNDERLYING_ACCESSOR_DECL(EnumVal, zeek_int_t, AsEnum)
```

Comment:
That's making it at least consistent with EnumType->Lookup(). Maybe it'll start warning/errors on `int y = x->AsEnum()` constructs, but that would be for the better.

@timwoj - pinging you explicitly. Mark as resolved if there are no concerns.

Reply:
@timwoj I think the pending issues are resolved with the push I just did, other than double-checking on this issue.

Reply:
Sorry, I must have missed the original ping on this one. I'm fine with this. There's so many places in this code where we implicitly cast from larger types to smaller ones.